dependencies/boost/meson.build

```diff
@@ -0,0 +1,22 @@
+# Boost Dependency File
+#============================================================================
+
+
+# Look for boost, a required dependency.
+#----------------------------------------------------
+boost_mod_list = [
+  'system',
+  'context',
+  'thread']
+
+
+boost_dep = declare_dependency(
+    link_args : get_option('boost_link_args'),
+    dependencies : [ dependency(
+        'boost',
+        modules: boost_mod_list)])
```

Comment:
These parentheses look much lispy. I don't think it is a good style for `meson.build` files.